### DISCUSSION TITLE
can someone give detailed information how to setup with ide autocomplete macros intelisense?

### DISCUSSION THREAD

**Discussion by deiiviidas** [GoogleFeud/ts-macros#93] [Q&A] [answered]
hello, body is the same as title - i can run macros, but i cant get any autocompletes/intelisenses from my ide... for macros i created/used im using webstorm, and i want to make next step to my programming AND elemininate boilerplate code/aka copy paste code.. i just cant get the idea how to setup with intelisenses.. i prefer vite/vue tech stack, with ts support. thank you if you wiling to help

do i need to declare them in .d.ts files? or it can be auto generated ? i dont understand

**Comment by deiiviidas** [comment]
so also it should got autocomplete with/for macros - like it should, but im missing it or i need make myself them? like this

https://github.com/GoogleFeud/ts-macros/wiki/Chaining-Macros
![image](https://github.com/user-attachments/assets/e79c270b-1c0e-4534-8f27-dba939302933)

**Reply by deiiviidas** [reply]
@GoogleFeud up?=

**Comment by volenslavchev** [comment, marked as answer]
Hi, if you're trying to set up the [type resolver](https://github.com/GoogleFeud/ts-macros/wiki/Type-Resolver-Transformer), I don't recommend it unless macros are an integral part of your project - it's still very experimental. I haven't tested it with the latest version of typescript.

If you want to get intellisense for chaining macros, yeah you'll have to type them up in type declarations yourself. [Here are some examples](https://googlefeud.github.io/ts-macros?code=FAehAIEEBNoSwHYHNwENwFtUGMBOB7caAUwDNE4AXOfBcSw9SgTwAdjgX3wAVYgZ0rgAvOADewcFPAA3VABsArsQBc4BIowAjYrknSwvABbFMOAvqkASLHnwB+ABQBKcGo3bdwAL7BQEHhMzOyIyRGpaYFJFBGwIuhtzfEc5eTU+QVcJaXBcYkpFXDpUgDpU5XAAanAARgBuHz9DAGE81EpTSiDy01QEaHBsBXl6INsLbFpBTGYMynSBIVExWQVlNRqAJgBmcF9JhH58eWIS+XwkRwxZxZLEuwBCF2cG-3AAMXxccC1FOHlKABaRD0NgCNCKJAYYgIIRdTpg4AkbDyVB5cBIc5aBTiSzgYgAD1YXyE4V0pBwpgAypRcIgUNkctIrIgSATIJRHKz3JodLhXGpBHTkA0cr5fG8ALJJULkBBUGiHKIxOKK8As-qEjmOIWC2n0gA04G56l5ugF4CF9NxOTyBSKltpAG1WQBdRpvZrDUamcb4YAHI4nM4XRwAIlQWmwYburK1lCeNWczj8hOJuCEYm8QA)


I hope this helps!

**Reply by deiiviidas** [reply]
can you please maybe share all examples(even hidden/not shared/not included ones)? is there any list of them? 

**that is very good source for learning;** becouse im like - i learned to code by coding, not reading or understanding how things works; no theory, but practical more - know how;
